libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/MeasurementUtils.java

```diff
@@ -0,0 +1,62 @@
+package com.mapbox.services.android.navigation.v5.utils;
+
+import com.mapbox.directions.v5.models.LegStep;
+import com.mapbox.geojson.Feature;
+import com.mapbox.geojson.LineString;
+import com.mapbox.geojson.Point;
+import com.mapbox.services.constants.Constants;
+import com.mapbox.services.utils.TextUtils;
+import com.mapbox.turf.TurfMeasurement;
+import com.mapbox.turf.TurfMisc;
+
+import static com.mapbox.turf.TurfConstants.UNIT_METERS;
+
+public final class MeasurementUtils {
+
+  private MeasurementUtils() {
+    // Util classes aren't allowed to be initialized.
+  }
+
+  /**
+   * Calculates the distance between the users current raw {@link android.location.Location} object
+   * to the closest {@link Point} in the {@link LegStep}.
+   *
+   * @param usersRawLocation {@link Point} the raw location where the user is currently located
+   * @param step             {@link LegStep} to calculate the closest point on the step to our
+   *                         predicted location
+   * @return double in distance meters
+   * @since 0.2.0
+   */
+  public static double userTrueDistanceFromStep(Point usersRawLocation, LegStep step) {
+    // Check that the leg step contains geometry.
+    if (TextUtils.isEmpty(step.geometry())) {
+
+      return 0;
+    }
+
+    // Get the lineString from the step geometry.
+    LineString lineString = LineString.fromPolyline(step.geometry(), Constants.PRECISION_6);
+
+    // Make sure that the step coordinates isn't empty. If the points equal each other, the distance
+    // is obviously zero, so return 0.
+    if (lineString.coordinates().isEmpty()
+      || usersRawLocation.equals(lineString.coordinates().get(0))) {
```

Comment:
Shouldn’t `TurfMeasurement.distance()` be responsible for checking whether the two coordinates are equal? Or does this get around a performance hit from calling `TurfMisc.pointOnLine()` unnecessarily?

Reply:
Looking deeper into this, Checking the two Point objects for equality does, in fact, prevent a performance hit. If we removed this check, the end result will be the same, a distance of 0 with no runtime exceptions being thrown. [This test](https://github.com/mapbox/mapbox-java/blob/cbec9e10d98e6676bbfcfc71a69462e5ff5007ff/services-turf/src/test/java/com/mapbox/turf/TurfMiscTest.java#L139-L151) confirms this.